src/overlays/ovl_i6/sf_i6_2.c

```diff
@@ -1,17 +1,474 @@
+                    otherActor->obj.pos.z = D_i6_801A68B0[i - 1].z;
+                    otherActor->obj.rot.z = D_i6_801A68F8[i - 1];
+                    otherActor->unk_0F4.x = 340.0f;
+                }
+
+                D_80178340 = D_80178358 = 255;
+            }
+            func_800B63BC(player, 1);
+            /* fallthrough */
+        case 1:
+
+            for (otherActor = &gActors[1], i = 1; i < 4; i++, otherActor++) {
+                otherActor->unk_0B8 = 0;
+                otherActor->timer_0BC = 3;
+                if (actor->timer_0BC == 0) {
+                    otherActor->unk_0B8 = 2;
+                    actor->unk_0B8 = 2;
+                    player->state_1C8 = PLAYERSTATE_1C8_3;
+                    player->unk_014 = 0.0001f;
+                    D_80177838 = 80;
+                }
+            }
+            if (D_8015F928 == 80) {
+                func_800BA808(gMsg_ID_19010, RCID_FOX);
+            }
+            break;
+        case 2:
+            func_i6_801962F4(actor);
+            if (((D_800C9B4C + 100) < D_8015F928) && (gActors[4].obj.status == 0) && (gActors[5].obj.status == 0) &&
+                (gActors[6].obj.status == 0) && (gActors[7].obj.status == 0) && (actor->timer_0BE == 0)) {
+                actor->timer_0BE = 80;
+            }
+            if ((actor->timer_0BE == 1) && (player->state_1C8 != PLAYERSTATE_1C8_7)) {
+
+                for (i = 1; i < ARRAY_COUNT(gTeamShields); i++) {
+                    D_80177C38[i] = gSavedTeamShields[i];
+                    D_801778F0[i] = gSavedTeamShields[i];
+                    gSavedTeamShields[i] = gTeamShields[i];
+                }
+
+                player->state_1C8 = PLAYERSTATE_1C8_7;
+                player->unk_1D0 = 0;
+                D_800D3180[gCurrentLevel] = 2;
+                D_80177C94 = gGoldRingCount[0];
+                D_80177C9C = player->shields + 1;
+                D_80161A5C = D_80177CA4 = gHitCount;
+                D_80177CAC = player->wings.rightState;
+                D_80177CB4 = player->wings.leftState;
+                D_80177CBC = gRightWingHealth[0];
+                D_80177CC4 = gLeftWingHealth[0];
+            }
+            break;
+        case 3:
+            D_801779BC = 0;
+            if (D_8015F908 < 600) {
+                player->camEye.x += actor4->vel.x * 0.23f;
+                player->camEye.y += actor4->vel.y * 0.23f;
+                player->camEye.z += actor4->vel.z * 0.23f;
+            }
+            Math_SmoothStepToF(&player->camAt.x, actor4->obj.pos.x, 1.0f, 20000.0f, 0.0f);
+            Math_SmoothStepToF(&player->camAt.y, actor4->obj.pos.y, 1.0f, 20000.0f, 0.0f);
+            Math_SmoothStepToF(&player->camAt.z, actor4->obj.pos.z, 1.0f, 20000.0f, 0.0f);
+            Math_SmoothStepToF(&player->unk_034, 0, 0.1f, 0.2f, 0.0f);
+            if ((gControllerPress->button & 0x1000) || (D_8015F928 == (D_800C9B4C + 300))) {
```

Comment:
```suggestion
if ((gControllerPress->button & START_BUTTON) || (D_8015F928 == (D_800C9B4C + 300))) {

Reply:
omg i saw that on the scratch but forgot to check if we had a macro 🤦🏼